jep/0000/README.adoc

```diff
@@ -0,0 +1,124 @@
+:caution-caption: :fire:
+:warning-caption: :warning:
+endif::[]
+
+.Metadata
+[cols="1h,1"]
+|===
+| JEP
+| 0000
+
+| Title
+| FIPS-140 Support
+
+| Sponsor
+| link:https://github.com/jtnord
+
+// Use the script `set-jep-status <jep-number> <status>` to update the status.
+| Status
+| Not Submitted :information_source:
+
+| Type
+| Standards
+
+| Created
+| Date (2023-09-06)
+
+| BDFL-Delegate
+| TBD
+
+//
+//
+// Uncomment if there is an associated placeholder JIRA issue.
+// the following is just one Jira - there may be more so not spcifically linking
+//| JIRA
+//| https://issues.jenkins-ci.org/browse/JENKINS-71971[JENKINS-71971]
+//
+//
+// Uncomment if discussion will occur in forum other than jenkinsci-dev@ mailing list.
+//| Discussions-To
+//| Link to where discussion and final status announcement will occur
+//
+//
+// Uncomment if this JEP depends on one or more other JEPs.
+//| Requires
+//| JEP-NUMBER, JEP-NUMBER...
+//
+//
+// Uncomment and fill if this JEP is rendered obsolete by a later JEP
+//| Superseded-By
+//| JEP-NUMBER
+//
+//
+// Uncomment when this JEP status is set to Accepted, Rejected or Withdrawn.
+//| Resolution
+//| Link to relevant post in the jenkinsci-dev@ mailing list archives
+
+|===
+
+== Abstract
+
+This proposal makes it possible to run Jenkins in such a way that it would be link:https://csrc.nist.gov/pubs/fips/140-2/upd2/final[FIPS-140] compliant.  Not all features may be possible to use in this mode of operation, and plugins may need to change their behaviour at runtime.
+Configuration of the JVM, platform etc. is out-of-scope, as is making every feature/plugin compliant.
+
+
+== Specification
+
+* A new API will be introduced that code can use to check if FIPS-140 compliance is preferred.
+* Code in Jenkins core (where necessary) switches behaviour between more secure yet non compliant cryptography to FIPS-140 compliant cryptography.
+* Non compliant and less secure crytpography has been replaced by a secure and compliant option.
+* General guidelines exist to help developers write FIPS-140 compliant code and/or understand incompatibilities with the standard.
+* User facing documentation provides a mere overview of the FIPS flag and what it does (and does not do).
+
+== Motivation
+
+Currently it is not possible to run Jenkins in a link:https://csrc.nist.gov/pubs/fips/140-2/upd2/final[FIPS-140-2] compliant way.  Many users needing Jenkins are required (e.g. for US government projects) to have any solution be compliant with FIPS-140 and today it is not possible for them to do this. 
+
+Both Jenkins core and plugins use cryptography that is not allowed and/or are not from an approved implementation, and this prevents adoption of Jenkins by these end users.
+Examples would be during setup where the install wizard creates a first admin user using the internal security realm which uses ``bcrypt`` which would fall fowl of the standard due to its use of ``blowfish`` which is not an approved algorithm.
```

Comment:
Minor spelling error, at least according to the [Merriam-Webster dictionary](https://www.merriam-webster.com/dictionary/fall%20foul%20of).  Non-blocking change.
```suggestion
Examples would be during setup where the install wizard creates a first admin user using the internal security realm which uses ``bcrypt`` which would fall foul of the standard due to its use of ``blowfish`` which is not an approved algorithm.
```